src/rdb.c

```diff
@@ -2719,31 +2733,15 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, redisDb* db, int rdbflags,
                 /* for TTL listpack, find the minimum expiry */
                 uint64_t minExpire = hashTypeGetNextTimeToExpire(o);
 
-                /* check if need to convert to dict encoding */
-                if ((db != NULL) &&
-                    (hashTypeLength(o, 0) > server.hash_max_listpack_entries))  /* TODO: each field length is not verified against server.hash_max_listpack_value */
-                {
-                    hashTypeConvert(o, OBJ_ENCODING_HT, &db->hexpires);
-                    /*
-                     * hashTypeAddToExpires is presumably called from within
-                     * the convert function (from listpackEx to dict), BUT,
-                     * this call depends on the lpt->meta field to be updated,
-                     * which is not the case here as hashTypeAddToExpires was
-                     * not yet called for the listpack (which is what updating
-                     * its meta).
-                     * Instead, this "manual" call is added here.
-                     * Another approach would be to have the conversion function
-                     * find the minExpire by itself when iterating on the listpack
-                     * instead of relying on the meta and use this value for the
-                     * final ebAdd call.
-                     */
+                /* Convert listpac to hash table without register in global HFE DS
+                 * since the listpack is not connected yet to the DB */
+                if (hashTypeLength(o, 0) > server.hash_max_listpack_entries)
```

Comment:
i realize that we forget to check if `hashTypeLength(o, 0)` returns 0, in this case, we should skip it and replication to slave like `hashmd`.

Reply:
i can't reproduce it yet, mark as resolved.